python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -0,0 +1,1150 @@
+        default=0.0
+    )
+    modelComponents = pexConfig.ListField(
+        dtype=str,
+        doc=("List of mappings to apply to transform from pixels to sky, in order of their application."
+             "Supported options are 'INSTRUMENT/DEVICE' and 'EXPOSURE'"),
+        default=["INSTRUMENT/DEVICE", "EXPOSURE"]
+    )
+    deviceModel = pexConfig.ListField(
+        dtype=str,
+        doc=("List of mappings to apply to transform from detector pixels to intermediate frame. Map names"
+             "should match the format 'BAND/DEVICE/<map name>'"),
+        default=["BAND/DEVICE/poly"]
+    )
+    exposureModel = pexConfig.ListField(
+        dtype=str,
+        doc=("List of mappings to apply to transform from intermediate frame to sky coordinates. Map names"
+             "should match the format 'EXPOSURE/<map name>'"),
+        default=["EXPOSURE/poly"]
+    )
+    devicePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of device polynomial model.",
+        default=4
+    )
+    exposurePolyOrder = pexConfig.Field(
+        dtype=int,
+        doc="Order of exposure polynomial model.",
+        default=6
+    )
+    fitProperMotion = pexConfig.Field(
+        dtype=bool,
+        doc="Fit the proper motions of the objects.",
+        default=False
+    )
+    excludeNonPMObjects = pexConfig.Field(
+        dtype=bool,
+        doc="Exclude reference objects without proper motion/parallax information.",
+        default=True
+    )
+
+    def setDefaults(self):
+        # Use only stars because aperture fluxes of galaxies are biased and
+        # depend on seeing.
+        self.sourceSelector["science"].doUnresolved = True
+        self.sourceSelector["science"].unresolved.name = "extendedness"
+
+        # Use only isolated sources.
+        self.sourceSelector["science"].doIsolated = True
+        self.sourceSelector["science"].isolated.parentName = "parentSourceId"
+        self.sourceSelector["science"].isolated.nChildName = "deblend_nChild"
+        # Do not use either flux or centroid measurements with flags,
+        # chosen from the usual QA flags for stars.
+        self.sourceSelector["science"].doFlags = True
+        badFlags = ["pixelFlags_edge",
+                    "pixelFlags_saturated",
+                    "pixelFlags_interpolatedCenter",
+                    "pixelFlags_interpolated",
+                    "pixelFlags_crCenter",
+                    "pixelFlags_bad",
+                    "hsmPsfMoments_flag",
+                    f"{self.sourceFluxType}_flag",
+                    ]
+        self.sourceSelector["science"].flags.bad = badFlags
+
+    def validate(self):
+        super().validate()
+
+        # Check if all components of the device and exposure models are
+        # supported.
+        for component in self.deviceModel:
+            if not (('poly' in component.lower()) or ('identity' in component.lower())):
```

Comment:
I'm going to be "that guy" who mentions that this file has a mix of single-quotes and double-quotes.  Now is the time to pick one.  (I personally like single quotes, but black prefers double quotes; I leave it up to you to choose which format for these files.)

Reply:
Ok, I kept double quotes for documentation, but standardized to single quotes otherwise.